### DISCUSSION TITLE
What's the easiest way to access colours

### DISCUSSION THREAD

**Discussion by hmajid2301** [tailwindlabs/tailwindcss#1910] [Help] [answered]
Within my own JS code is there an interface Tailwind exposes to allow us to access a given colour's, such as `blue-500`, value.

I would like to be able to access the tailwind configuration combined with the user configuration within JS is this possible. ? Thanks

**Comment by estevanmaito** [comment, marked as answer]
Following the docs, it would look something like this: https://tailwindcss.com/docs/configuration/#referencing-in-javascript

```js
import resolveConfig from 'tailwindcss/resolveConfig'
import tailwindConfig from './tailwind.config.js'

const {theme} = resolveConfig(tailwindConfig)

theme.colors.blue[500]
```

**Reply by codinginflow** [reply]
How can I make this work if the colors reference CSS variables?
E.g. `theme.colors.primary.foreground` prints `hsl(var(--primary-foreground))`.

**Reply by wongjn** [reply]
Yes, that seems correct.

**Reply by codinginflow** [reply]
I mean, how can I get the actually hex value in this case?

**Reply by wongjn** [reply]
You'd need to resolve the value of `var(--primary-foreground)`. This could differ depending on the element you resolve from and any relevant CSS.

**Comment by 1mehdifaraji** [comment]
I did it very easily by importing `colors` from `tailwindcss/colors`

```javascript
import colors from "tailwindcss/colors";

colors.emerald["100"]
```
